packages/router-core/src/serializer.ts

```diff
@@ -1,32 +1,150 @@
+import type { Plugin } from 'seroval'
+import type { Constrain } from './utils'
+
 export interface StartSerializer {
   stringify: (obj: unknown) => string
   parse: (str: string) => unknown
   encode: <T>(value: T) => T
   decode: <T>(value: T) => T
 }
 
-export type SerializerStringifyBy<T, TSerializable> = T extends TSerializable
-  ? T
-  : T extends (...args: Array<any>) => any
-    ? 'Function is not serializable'
-    : { [K in keyof T]: SerializerStringifyBy<T[K], TSerializable> }
+// help me fill in the runtime
+export interface Serializer {
+  '~types': {
+    serializer: unknown
+  }
+  stringify: (obj: unknown) => string
+  parse: (str: string) => unknown
+}
+
+export type TypeSerializerStringifyBy<T, TSerializable> =
+  T extends TSerializable
+    ? T
+    : T extends (...args: Array<any>) => any
+      ? 'Function is not serializable'
+      : { [K in keyof T]: TypeSerializerStringifyBy<T[K], TSerializable> }
 
-export type SerializerParseBy<T, TSerializable> = T extends TSerializable
+export type TypeSerializerParseBy<T, TSerializable> = T extends TSerializable
   ? T
   : unknown extends SerializerExtensions['ReadableStream']
-    ? { [K in keyof T]: SerializerParseBy<T[K], TSerializable> }
+    ? { [K in keyof T]: TypeSerializerParseBy<T[K], TSerializable> }
     : T extends SerializerExtensions['ReadableStream']
       ? ReadableStream
-      : { [K in keyof T]: SerializerParseBy<T[K], TSerializable> }
+      : { [K in keyof T]: TypeSerializerParseBy<T[K], TSerializable> }
 
 export interface DefaultSerializerExtensions {
   ReadableStream: unknown
 }
 
 export interface SerializerExtensions extends DefaultSerializerExtensions {}
 
-export type Serializable = Date | undefined | Error | FormData | bigint
+export type DefaultTypeSerializable =
```

Comment:
what makes it the default? that it is supported by seroval without any plugin?

Reply:
Its just the existing serialization code, we can probably remove it